TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/PATCH-Positive-Deviceresource.robot

```diff
@@ -4,39 +4,53 @@ Resource     TAF/testCaseModules/keywords/core-metadata/coreMetadataAPI.robot
-    And deviceResource Should Be Updated
-    [Teardown]  Delete Device Profile By Name
+    And Profile Test-Profile-1 Resource "description" Should Be Updated
+    And Profile Test-Profile-1 Resource "isHidden" Should Be Updated
+    [Teardown]  Delete Device Profile By Name  Test-Profile-1
 
-ProfileResourcePATCH002 - Update multiple deviceResources on one device profile
-    Given Create a device profile with multiple device resources
-    When Update deviceResource
+ProfileResourcePATCH002 - Update multiple resources on multiple device profiles
+    # multiple resources > multiple profiles
+    Given Generate Multiple Device Profiles Sample
+    And Create Device Profile ${deviceProfile}
+    And Generate multiple resource sample for updating  Test-Profile-1  Test-Profile-2  Test-Profile-3
+    When Update resource ${resourceUpdate}
     Then Should Return Status Code "207"
     And Should Return Content-Type "application/json"
     And Item Index All Should Contain Status Code "200"
     And Response Time Should Be Less Than "${default_response_time_threshold}"ms
-    And deviceResource Should Be Updated
-    [Teardown]  Delete Device Profile By Name
+    And Profile Test-Profile-1 Resource "description" Should Be Updated
+    And Profile Test-Profile-1 Resource "isHidden" Should Be Updated
+    And Profile Test-Profile-2 Resource "description" Should Be Updated
+    And Profile Test-Profile-3 Resource "isHidden" Should Be Updated
+    [Teardown]  Delete Multiple Device Profiles By Names  Test-Profile-1  Test-Profile-2  Test-Profile-3
 
-ProfileResourcePATCH003 - Update multiple deviceResources on multiple device profiles
-    Given Create multiple device profiles with multiple device resources
-    When Update deviceResource
-    Then Should Return Status Code "207"
-    And Should Return Content-Type "application/json"
-    And Item Index All Should Contain Status Code "200"
-    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
-    And deviceResource Should Be Updated
-    [Teardown]  Delete Multiple Device Profiles By Names
+*** Keywords ***
+Generate multiple resource sample for updating
+    [Arguments]  ${test_profile_1}  ${test_profile_2}  ${test_profile_3}
+    ${resource_1}=  Create Dictionary  name=DeviceValue_Boolean_RW  description=Dcp_ABC  isHidden=${false}
+    ${resource_2}=  Create Dictionary  name=DeviceValue_String_R  description=Dcp_ABC
+    ${resource_3}=  Create Dictionary  name=DeviceValue_FLOAT32_R  isHidden=${false}
+    Generate updating resource  ${resource_1}  ${resource_2}  ${resource_3}
+    Set To Dictionary  ${resourceUpdate}[0]  profileName=Test-Profile-1
+    Set To Dictionary  ${resourceUpdate}[1]  profileName=Test-Profile-2
+    Set To Dictionary  ${resourceUpdate}[2]  profileName=Test-Profile-3
+
+Profile ${test_profile} Resource "${property}" Should Be Updated
+    Query device profile by name  ${test_profile}
```

Comment:
It would be better to use GET `/deviceresource/profile/{profileName}/resource/{resourceName}` to check updates.

Reply:
Thanks for the tips.
I've revised the script and used GET` /deviceresource/profile/{profileName}/resource/{resourceName}` to check updates. 
I tried to put them in one keyword to verify the update but in vain so I separated them into two keywords for now.
Tc001 will fail in all-in-one keyword, for it cannot find test-profile-2 and test-profile-3.